data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -90,31 +102,12 @@ public GrokPrepper(final PluginSetting pluginSetting) {
         for (final Record<String> record : records) {
             try {
                 final Map<String, Object> recordMap = OBJECT_MAPPER.readValue(record.getData(), MAP_TYPE_REFERENCE);
-                final Map<String, Object> grokkedCaptures = new HashMap<>();
-
-                for (final Map.Entry<String, List<Grok>> entry : fieldToGrok.entrySet()) {
-                    for (final Grok grok : entry.getValue()) {
-                        if (recordMap.containsKey(entry.getKey())) {
-                            final Match match = grok.match(recordMap.get(entry.getKey()).toString());
-                            match.setKeepEmptyCaptures(grokPrepperConfig.isKeepEmptyCaptures());
-
-                            final Map<String, Object> captures = match.capture();
-                            mergeCaptures(grokkedCaptures, captures);
-
-                            if (shouldBreakOnMatch(grokkedCaptures)) {
-                                break;
-                            }
-                        }
-                    }
-                    if (shouldBreakOnMatch(grokkedCaptures)) {
-                        break;
-                    }
-                }
 
-                if (grokPrepperConfig.getTargetKey() != null) {
-                    recordMap.put(grokPrepperConfig.getTargetKey(), grokkedCaptures);
+                if (grokPrepperConfig.getTimeoutMillis() == 0) {
+                    matchAndMerge(recordMap);
                 } else {
-                    mergeCaptures(recordMap, grokkedCaptures);
+                    final Runnable matchTimeoutRunnable = () -> matchAndMerge(recordMap);
```

Comment:
You should be able to put the lambda directly in `runWithTimeout` to remove this extra variable. And it is somewhat more common to see lambdas passed into methods.